src/Http/Headers/src/HeaderUtilities.cs

```diff
@@ -696,25 +696,25 @@ public static StringSegment EscapeAsQuotedString(StringSegment input)
             // By calling this, we know that the string requires quotes around it to be a valid token.
             var backSlashCount = CountAndCheckCharactersNeedingBackslashesWhenEncoding(input);
 
-            var stringBuilder = new InplaceStringBuilder(input.Length + backSlashCount + 2); // 2 for quotes
-            stringBuilder.Append('\"');
+            return string.Create(input.Length + backSlashCount + 2, input, (span, segment) => {
+                span[0] = span[span.Length - 1] = '\"';
```

Comment:
```suggestion
                span[span.Length - 1] = span[0] = '\"';
```

In the current code there will be a bounds check for `span[0]` and one for `span[span.Length - 1]`.
With the suggestion there will be only one bounds check for `span[span.Length - 1]`.

Reply:
Could we add a comment with the code if we accept these suggestions? They're super non-obvious

Reply:
I've added a comment to the suggestion.

Ideally the JIT would figure out this by himself, but this is not the case as of today.